### ISSUE TITLE
Support for Jupyter notebooks

### ISSUE THREAD

**Issue by Matelasse** [lapce/lapce#1037]
This would probably require lots of UI and core effort, but it would be a great feature, especially considering the fact that there are no native GUI notebook editors. (there is vscode, but it's electron)

**Comment by TigerbyteDev** [comment]
> This would probably require lots of UI and core effort, but it would be a great feature, especially considering the fact that there are no native GUI notebook editors. (there is vscode, but it's electron)

There are also some Jetbrains IDEs with it in case you like them :)

**Comment by jerabaul29** [comment]
Agree this would be awesome - jupyter notebooks are becoming very commonplace in several communities. Any plans on this? :)

**Comment by panekj** [comment]
> Any plans on this? :)

The plan is that if there will be time and resources, it might be implemented.

**Comment by mrvruj** [comment]
As a data scientist, I would absolutely love a non-Electron, native program with Jupyter support. I'd be willing to put a $100 bounty on this -- paltry probably for the time it would take to develop, but it would be something at least. Not sure if this project accepts bounties or not -- in any case, thanks for making such a great tool!

**Comment by MinusGix** [comment]
We don't currently have any form of bounty system set up (partially due to lack of bounties, though not having an obvious one setup makes them less likely). I think we're currently unsure whether to put it into the editor, to add it as a plugin. Adding it into the editor itself would be faster, but plugin would be more 'natural' - but would also require our plugin system to be significantly more fleshed out. Also lots of other features that always feel like more of a focus.

**Comment by alex180500** [comment]
If this is added I would make the switch instantly and I think many more would follow, for sure in the scientific computation community. I mostly use jupyter notebook for Python but more and more people use them also for Julia.

**Comment by abstractqqq** [comment]
I agree with everyone else and I am willing to put down a small bounty as well

**Comment by panekj** [comment]
The problem with bounties is that someone pays money, someone receives it but it ultimately comes to us supporting the code that is being written and if there arises a need of major refactor (like what happened with Druid/Floem) such code might break or simply get deleted.

I know that everyone would want their feature to be implemented NOW but we have to work on editor stability and basic features before we get to such big functionalities.

**Comment by Ghos3t** [comment]
> We don't currently have any form of bounty system set up (partially due to lack of bounties, though not having an obvious one setup makes them less likely). I think we're currently not 100% sure whether to integrate into the editor, or to add it as a plugin. Adding it into the editor itself would be faster to implement, but a plugin would be more 'natural' - however it would also require our plugin system to be significantly more fleshed out. Also lots of other features that always feel like more of a focus.

A plugin that supports working on Jupyter Notebooks similar to VSCode would be great. A more robust plugin ecosystem itself could be a game changer for Lapce, I think this is one area that helped popularize VSCode.

**Comment by teanijarv** [comment]
+1 on this request
